### ISSUE TITLE
Add Bert Model

### ISSUE THREAD

**Issue by sam5658** [Rakesh9100/ML-Project-Drug-Review-Dataset#20]
### Prerequisites

- [X] I am an Open Source Contributor
- [X] I checked to make sure that this issue has not already been filed

### Description

Hi
I'm Shubham | Contributor GSSOC'23.
Many models are available for performing sentiment analysis(in our case, patient review), which has proven to cause a significant rise in prediction accuracy. Therefore I'm willing to implement the BERT model which can be utilized with the help of the hugging face "transformer" library.

**Implementation:**

- Deep Learning Models such as Bert are known for their great Transformer based Architecture, Bert's architecture needs to be fine-tuned on our Drug Dataset.
- hugging face transformers have various functionalities which can ease our model training keeping the code readable.
- I suggest training the model with the help of Pytorch library(only for creating CustomDataset) and DataLoaders
- All of these points are needed to be coded in a different file than "main.py" so that those unfamiliar with the PyTorch library can work still go through the code base.

**Deliverables:**

- A separate file from "main.py" implementing and training the BERT model for improving the accuracy.
- The deep learning model can then be imported into the file "main.py" with just a few lines of code.

**issue**
Further for visualisation, I suggest using seaborn only as sklearn.metrics.plot_confusion_matrics is deprecated.

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/Rakesh9100/ML-Project-Drug-Review-Dataset/blob/main/.github/CONTRIBUTING_GUIDELINES.md) of this project.

**Comment by Rakesh9100** [comment]
@sam5658 Okay Assigning you, Start working

**Comment by Rakesh9100** [comment]
@sam5658 Status of your work

**Comment by sam5658** [comment]
The data set was quite big, that's why I couldn't train the model as it was
reading the entire CPU ram, but the code has been optimized now and I will
put the model in training and give a pull request in a couple of days.

**Comment by Rakesh9100** [comment]
@sam5658 Update pls

**Comment by Rakesh9100** [comment]
Closing for having no response.
